subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/rmf/RMFSubscriptionFreeTrialActiveMatchingAttribute.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.subscriptions.impl.rmf
+
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.remote.messaging.api.AttributeMatcherPlugin
+import com.duckduckgo.remote.messaging.api.JsonMatchingAttribute
+import com.duckduckgo.remote.messaging.api.JsonToMatchingAttributeMapper
+import com.duckduckgo.remote.messaging.api.MatchingAttribute
+import com.duckduckgo.subscriptions.api.ActiveOfferType
+import com.duckduckgo.subscriptions.impl.SubscriptionsManager
+import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = JsonToMatchingAttributeMapper::class,
+)
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = AttributeMatcherPlugin::class,
+)
+@SingleInstanceIn(AppScope::class)
+class RMFSubscriptionFreeTrialActiveMatchingAttribute @Inject constructor(
+    private val subscriptionsManager: SubscriptionsManager,
+) : JsonToMatchingAttributeMapper, AttributeMatcherPlugin {
+    override suspend fun evaluate(matchingAttribute: MatchingAttribute): Boolean? {
+        return when (matchingAttribute) {
+            is SubscriptionFreeTrialActiveMatchingAttribute -> {
+                val hasActiveTrial = subscriptionsManager.getSubscription()
```

Comment:
@nalcalag why not using `suspend fun isFreeTrialActive(): Boolean` from `AuthRepository`. It relies on the value stored in `SubscriptionsDataStore`.

I'm curious because I'm using that same function in other places to check the same.